wcfsetup/install/files/lib/form/RegisterActivationForm.class.php

```diff
@@ -77,13 +77,13 @@ public function validate() {
 			throw new UserInputException('username', 'notFound');
 		}
 		
-		// user is already enabled
-		if ($this->user->activationCode == 0) {
+		// user email is already confirmed
+		if ($this->user->isEmailConfirmed()) {
 			throw new NamedUserException(WCF::getLanguage()->get('wcf.user.registerActivation.error.userAlreadyEnabled'));
 		}
 		
 		// check given activation code
-		if ($this->user->activationCode != $this->activationCode) {
+		if (!\hash_equals($this->activationCode, $this->user->emailConfirmed)) {
```

Comment:
Switch the parameter order of hash equals. The first parameter is defined to be the `known_string`.
```suggestion
		if (!\hash_equals($this->user->emailConfirmed, $this->activationCode)) {
```